nistats/datasets.py

```diff
@@ -237,8 +237,68 @@ def _glob_spm_multimodal_fmri_data():
             _uncompress_file(archive_path)
         except:
             print("Archive corrupted, trying to download it again.")
-            return fetch_spm_multimodal_fmri_data(data_dir=data_dir,
-                                                  data_name="",
-                                                  subject_id=subject_id)
+            return fetch_spm_multimodal_fmri(data_dir=data_dir,
+                                             data_name="",
+                                             subject_id=subject_id)
 
     return _glob_spm_multimodal_fmri_data()
+
+
+def fetch_fiac_first_level(data_dir=None, verbose=1):
+    """ Download a first-level fiac fMRI dataset (2 sessions)
+
+    Parameters
+    ----------
+    data_dir: string
+        directory where data should be downloaded and unpacked.
+    """
+    data_dir = _get_dataset_dir('', data_dir=data_dir, verbose=verbose)
+
+    def _glob_fiac_data():
+        """glob data from subject_dir."""
+        _subject_data = {}
+        subject_dir = os.path.join(data_dir, 'nipy-data-0.2/data/fiac/fiac0')
+        for session in [1, 2]:
+            # glob func data for session s + 1
+            session_func = os.path.join(subject_dir, 'run%i.nii.gz' % session)
+            if not os.path.isfile(session_func):
+                print('Missing functional scan for session %i.' % session)
+                return None
+            else:
+                _subject_data['func%i' % session] = session_func
+
+            # glob design matrix .npz file
+            sess_dmtx = os.path.join(subject_dir, 'run%i_design.npz' % session)
+            if not os.path.isfile(sess_dmtx):
+                print('Missing session file: %s' % sess_dmtx)
+                return None
+            else:
+                _subject_data['design_matrix%i' % session] = sess_dmtx
+
+        # glob for mask data
+        mask = os.path.join(subject_dir, 'mask.nii.gz')
+        if not os.path.isfile(mask):
+            print('Missing mask image.')
+            return None
+        else:
+            _subject_data['mask'] = mask
+        return Bunch(**_subject_data)
+
+    # maybe data_dir already contains the data ?
+    data = _glob_fiac_data()
+    if not data is None:
+        return data
+
+    # No. Download the data
+    print('Data absent, downloading...')
+    url = 'http://nipy.sourceforge.net/data-packages/nipy-data-0.2.tar.gz'
+
+    archive_path = os.path.join(data_dir, os.path.basename(url))
+    _fetch_file(url, data_dir)
+    try:
+        _uncompress_file(archive_path)
+    except:
+        print('Archive corrupted, trying to download it again.')
+        return fetch_fiac_first_level(data_dir=data_dir)
```

Comment:
Just curious, why do you need both fetch_fiac_first_level and _glob_fiac_data, is that to have a quick and dirty retry mechanism?